src/components/PostDetails.tsx

```diff
@@ -1,107 +1,75 @@
-import React from 'react';
-import { Loader } from './Loader';
-import { NewCommentForm } from './NewCommentForm';
+import React, { useEffect, useState } from 'react';
 
-export const PostDetails: React.FC = () => {
-  return (
-    <div className="content" data-cy="PostDetails">
-      <div className="content" data-cy="PostDetails">
-        <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
-
-          <p data-cy="PostBody">
-            eveniet quo quis laborum totam consequatur non dolor ut et est
-            repudiandae est voluptatem vel debitis et magnam
-          </p>
-        </div>
-
-        <div className="block">
-          <Loader />
+import { Loader, CommentItem, NewCommentForm } from './index';
+import { Post } from '../types';
+import { useComments } from '../hooks/useComments';
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+type Props = {
+  selectedPost: Post;
+};
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+export const PostDetails: React.FC<Props> = ({ selectedPost }) => {
+  const { id, title, body } = selectedPost;
 
-          <p className="title is-4">Comments:</p>
+  const [isFormOpened, setIsFormOpened] = useState(false);
+  const { comments, errorMessage, isLoading, deleteComment, addComment } =
+    useComments(id);
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+  useEffect(() => setIsFormOpened(false), [selectedPost]);
```

Comment:
The `useEffect` hook resets the form state when the `selectedPost` changes. Ensure that this behavior is intended, as it will close the comment form whenever a new post is selected.